blockchain/state_manager/src/lib.rs

```diff
@@ -973,28 +969,43 @@ where
                     _ => (),
                 }
             }
-
             Ok((None, None))
-        });
-
-        futures.push(subscriber_poll);
-        let search_back_poll = task::spawn(async move {
-            let (back_tipset, back_receipt) = task.await?;
-            let should_revert = *reverts
-                .read()
-                .await
-                .get(back_tipset.key())
-                .unwrap_or(&false);
-            let larger_height_of_head = height_of_head >= back_tipset.epoch() + confidence;
-            if !should_revert && larger_height_of_head {
-                return Ok((Some(back_tipset), Some(back_receipt)));
+        })
+        .fuse();
+
+        let mut search_back_poll = task::spawn::<
+            _,
+            Result<(Option<Arc<Tipset>>, Option<MessageReceipt>), Error>,
+        >(async move {
```

Comment:
Optional: only the error type needs to be inferred

```suggestion
        let mut search_back_poll = task::spawn::<_, Result<_, Error>>(async move {
```